File: agents/corpus_benchmark/p1_validations/validation.php
<?php
function validatePassword($password) {
    if (strlen($password) < 8) {
        throw new Exception("Mot de passe trop court");
    }
    if (!preg_match("/[A-Z]/", $password)) {
        throw new Exception("Majuscule requise");
    }
    return true;
}

function validerEmail($email) {
    if (!filter_var($email, FILTER_VALIDATE_EMAIL)) {
        throw new Exception("Email invalide");
    }
    return true;
}

function validerAge($age) {
    if (!is_numeric($age)) {
        throw new Exception("Age non numerique");
    }
    if ($age < 18) {
        throw new Exception("Doit etre majeur");
    }
    return true;
}

File: agents/benchmark_resultats/sans_patrons/sorties/p1_validations/validation.py
"""
Migré automatiquement par SMAML depuis validation.php
"""

# ── validate_password (score 92.0%, 1 itération(s)) ──
from pydantic import ValidationError
from fastapi import HTTPException
from typing import Optional

def validate_password(password: str) -> bool:
    if len(password) < 8:
        raise HTTPException(status_code=400, detail="Mot de passe trop court")
    if not any(char.isupper() for char in password):
        raise HTTPException(status_code=400, detail="Majuscule requise")
    return True


# ── valider_email (score 76.0%, 2 itération(s)) ──
from pydantic import EmailStr
from fastapi import HTTPException

def valider_email(email: str) -> bool:
    try:
        email_valid = EmailStr.validate(email)
        return True
    except ValueError:
        raise HTTPException(status_code=400, detail="Email invalide")


# ── valider_age (score 88.4%, 1 itération(s)) ──
from pydantic import BaseModel
from fastapi import HTTPException

class Age(BaseModel):
    age: int

def valider_age(age: int) -> bool:
    if not isinstance(age, int):
        raise HTTPException(status_code=400, detail="Age non numérique")
    if age < 18:
        raise HTTPException(status_code=400, detail="Doit être majeur")
    return True
